src/libcore/clone.rs

```diff
@@ -63,28 +63,39 @@
 /// This trait can be used with `#[derive]` if all fields are `Clone`. The `derive`d
 /// implementation of [`clone`] calls [`clone`] on each field.
 ///
+/// For a generic struct, `#[derive]` implements `Clone` conditionally by adding bound `Clone` on
+/// generic parameters.
+///
+/// ```
+/// // After adding the derive, Reading<T> is Clone if T is Clone.
```

Comment:
Do we make this even easier to understand if we say "when T is Clone"? Since it's a parameter that varies, maybe this clarifies it for some readers?